dot/parachain/candidate-validation/candidate_validation.go

```diff
@@ -73,67 +78,56 @@ func (*CandidateValidation) ProcessBlockFinalizedSignal(parachaintypes.BlockFina
 
 // Stop stops the CandidateValidation subsystem
 func (cv *CandidateValidation) Stop() {
-	close(cv.stopChan)
-	cv.wg.Wait()
 }
 
 // processMessages processes messages sent to the CandidateValidation subsystem
-func (cv *CandidateValidation) processMessages(wg *sync.WaitGroup) {
-	defer wg.Done()
-	for {
-		select {
-		case msg := <-cv.OverseerToSubsystem:
-			logger.Debugf("received message %v", msg)
-			switch msg := msg.(type) {
-			case ValidateFromChainState:
-				runtimeInstance, err := cv.BlockState.GetRuntime(msg.CandidateReceipt.Descriptor.RelayParent)
-				if err != nil {
-					logger.Errorf("failed to get runtime: %w", err)
-					msg.Ch <- parachaintypes.OverseerFuncRes[ValidationResult]{
-						Err: err,
-					}
-				}
-				result, err := validateFromChainState(runtimeInstance, msg.Pov, msg.CandidateReceipt)
-				if err != nil {
-					logger.Errorf("failed to validate from chain state: %w", err)
-					msg.Ch <- parachaintypes.OverseerFuncRes[ValidationResult]{
-						Err: err,
-					}
-				} else {
-					msg.Ch <- parachaintypes.OverseerFuncRes[ValidationResult]{
-						Data: *result,
-					}
-				}
-			case ValidateFromExhaustive:
-				result, err := validateFromExhaustive(cv.ValidationHost, msg.PersistedValidationData,
-					msg.ValidationCode, msg.CandidateReceipt, msg.PoV)
-				if err != nil {
-					logger.Errorf("failed to validate from exhaustive: %w", err)
-					msg.Ch <- parachaintypes.OverseerFuncRes[ValidationResult]{
-						Err: err,
-					}
-				} else {
-					msg.Ch <- parachaintypes.OverseerFuncRes[ValidationResult]{
-						Data: *result,
-					}
-				}
-
-			case PreCheck:
-				// TODO: implement functionality to handle PreCheck, see issue #3921
-
-			case parachaintypes.ActiveLeavesUpdateSignal:
-				_ = cv.ProcessActiveLeavesUpdateSignal(msg)
-
-			case parachaintypes.BlockFinalizedSignal:
-				_ = cv.ProcessBlockFinalizedSignal(msg)
-
-			default:
-				logger.Errorf("%w: %T", parachaintypes.ErrUnknownOverseerMessage, msg)
+func (cv *CandidateValidation) processMessages(msg any) {
+	switch msg := msg.(type) {
+	case ValidateFromChainState:
+		runtimeInstance, err := cv.BlockState.GetRuntime(msg.CandidateReceipt.Descriptor.RelayParent)
+		if err != nil {
```

Comment:
should you return inside this condition?

Reply:
Yes, Updated it